LTSupportAutomotive/LTVIN.m

```diff
@@ -52,8 +52,20 @@ -(void)decode
         _manufacturer = LTStringLookupWithPlaceholder( manufacturerCode2Digits, UNKNOWN );
     }
     
+    static NSUInteger const MODEL_YEAR_RANGE_DISCRIMINATOR_LOCATION = 6;
+    static NSUInteger const MODEL_YEAR_CODE_LOCATION = 9;
+    
+    if (isdigit([_vin characterAtIndex:MODEL_YEAR_RANGE_DISCRIMINATOR_LOCATION])) {
+        // if the character at index 6 is a numeric digit then we select our model year from the 'old' 1980-2009 range
+        NSString* modelYearCode = [@"VIN_MODEL_YEAR_OLD_" stringByAppendingString:[_vin substringWithRange:NSMakeRange(MODEL_YEAR_CODE_LOCATION, 1)]];
+        _modelYear = LTStringLookupWithPlaceholder( modelYearCode, UNASSIGNED );
+    } else {
+        // if the character at index 6 is not a numeric digit then we select our model year from the 'new' 2010-2029 range
+        NSString* modelYearCode = [@"VIN_MODEL_YEAR_NEW_" stringByAppendingString:[_vin substringWithRange:NSMakeRange(MODEL_YEAR_CODE_LOCATION, 1)]];
+        _modelYear = LTStringLookupWithPlaceholder( modelYearCode, UNASSIGNED );
+    }
+    
```

Comment:
Does that really hold for all kind of VINs?

Reply:
On Wikipedia page: https://en.wikipedia.org/wiki/Vehicle_identification_number in section "Model Year Encoding" it says that:
On April 30, 2008, the US National Highway Traffic Safety Administration adopted a final rule amending 49 CFR Part 565, "so that the current 17 character vehicle identification number (VIN) system, which has been in place for almost 30 years, can continue in use for at least another 30 years", in the process making several changes to the VIN requirements applicable to all motor vehicles manufactured for sale in the United States:
In order to identify the exact year in passenger cars and multipurpose passenger vehicles with a GVWR of 10,000 or less, one must read position 7 as well as position 10. For passenger cars, and for multipurpose passenger vehicles and trucks with a gross vehicle weight rating of 10,000 lb (4,500 kg) or less, if position seven is numeric, the model year in position 10 of the VIN refers to a year in the range 1980–2009. If position seven is alphabetic, the model year in position 10 of VIN refers to a year in the range 2010–2039.

Reply:
I cant guarantee this and it probably needs more testing. But without something like this, then the library will return incorrect model year for most recent cars. At some point in time in a modern app we should err on expecting that a car is more likely to be 2010-2039 than be 1980-2009. Tough call. Perhaps you could add support here for a heuristic or algorithm that looked at other elements in the VIN or in the other PIDs to help decide if the car is 1980-2009 or 2010-2039.

Reply:
I agree. Given also that this VIN class is probably most often used to identify car's attached to an OBD2 adapter, the model year almost always has to be above 1996.